crates/sui-framework/packages/sui-framework/sources/coin.move

```diff
@@ -160,6 +160,7 @@ public fun put<T>(balance: &mut Balance<T>, coin: Coin<T>) {
 
 // === Base Coin functionality ===
 
+#[allow(lint(public_entry))]
 /// Consume the coin `c` and add its value to `self`.
 /// Aborts if `c.value + self.value > U64_MAX`
 public entry fun join<T>(self: &mut Coin<T>, c: Coin<T>) {
```

Comment:
For removing this, maybe we should infer entry "cleanliness" in the PTB adapter rewrite